precompiles/proxy/src/lib.rs

```diff
@@ -0,0 +1,219 @@
+	/// The dispatch origin for this call must be Signed.
+	///
+	/// Parameters:
+	/// * delegate: The account that the caller would like to remove as a proxy.
+	/// * proxy_type: The permissions currently enabled for the removed proxy account.
+	/// * delay: The announcement period required of the initial proxy. Will generally be zero.
+	fn remove_proxy(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let mut input = handle.read_input()?;
+		input.expect_arguments(3)?;
+
+		let delegate = input
+			.read::<Address>()
+			.map(|addr| Runtime::AddressMapping::into_account_id(addr.into()))?;
+		let proxy_type = input
+			.read::<u8>()
+			.map_err(|_| revert("failed reading proxy_type"))
+			.and_then(|value| {
+				Runtime::ProxyType::decode(&mut value.to_le_bytes().as_slice())
+					.map_err(|_| revert("failed decoding proxy_type"))
+			})?;
+		let delay = input.read::<u32>()?.into();
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::remove_proxy {
+			delegate,
+			proxy_type,
+			delay,
+		}
+		.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Unregister all proxy accounts for the sender.
+	/// The dispatch origin for this call must be Signed.
+	/// WARNING: This may be called on accounts created by anonymous, however if done, then the
+	/// unreserved fees will be inaccessible. All access to this account will be lost.
+	fn remove_proxies(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::remove_proxies {}.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Checks if the caller has an account proxied with a given proxy type
+	///
+	/// Parameters:
+	/// * delegate: The account that the caller has maybe proxied
+	/// * proxyType: The permissions allowed for the proxy
+	fn is_proxy(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let mut input = handle.read_input()?;
+		input.expect_arguments(2)?;
+
+		let delegate = input
+			.read::<Address>()
+			.map(|addr| Runtime::AddressMapping::into_account_id(addr.into()))?;
+		let proxy_type = input
+			.read::<u8>()
+			.map_err(|_| revert("failed reading proxy_type"))
+			.and_then(|value| {
+				Runtime::ProxyType::decode(&mut value.to_le_bytes().as_slice())
+					.map_err(|_| revert("failed decoding proxy_type"))
+			})?;
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let is_proxy = ProxyPallet::<Runtime>::proxies(origin)
+			.0
+			.iter()
+			.any(|pd| pd.delegate == delegate && pd.proxy_type == proxy_type);
+
+		Ok(succeed(EvmDataWriter::new().write(is_proxy).build()))
```

Comment:
It would be good to return the proxy delay as well, because if we don't know the delay we can't remove the proxy.

Reply:
This function is currently only returning bool, should we accept delay are a filter parameter as well? Or modify the function to return a (bool, delay) tuple?

Aalternatively I'm not sure how the delay parameter would be much useful since it only says number of blocks and not a particular block number (i.e. deadline). Personally don't see much value returning it, especially since the precompile would be used to manage proxies for self and thus keep track of delay themselves anyhow.

Reply:
> I'm not sure how the delay parameter would be much useful since it only says number of blocks and not a particular block number

Let's imagine that a smart contract needs an authorization to do some things on my behalf.
The smart contract will need to check that the proxy exists, but also to check its delay, if it imposes a proxy without delay, or if it needs to behave differently depending on the delay.